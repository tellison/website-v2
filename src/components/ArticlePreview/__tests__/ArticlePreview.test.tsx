import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import ArticlePreview from '../index';

describe('ArticlePreview component', () => {
    it('should render correctly', () => {
        const { container } = render(
            <ArticlePreview
                author='Adoptium PMC'
                date='2021-01-01'
                postPath='/test-post'
                title='Test Title'
                description='Test Description'
                identifier='pmc'
            />
        );

        expect(container).toMatchSnapshot();
    });

    it('should render correctly no match', () => {
        const { container } = render(
            <ArticlePreview
                author='Test Author'
                date='2021-01-01'
                postPath='/test-post'
                title='Test Title'
                excerpt='Test Excerpt'
                identifier='foo'
            />
        );

        expect(container).toMatchSnapshot();
    });
});