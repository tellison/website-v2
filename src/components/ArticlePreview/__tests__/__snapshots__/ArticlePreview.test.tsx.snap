// Vitest Snapshot v1

exports[`ArticlePreview component > should render correctly 1`] = `
<div>
  <article
    class="pb-3"
  >
    <header>
      <h2
        class="h3"
      >
        <a
          href="/test-post"
        >
          Test Title
        </a>
      </h2>
      <div
        class="byline"
        style="display: flex; align-items: center; margin-top: 1rem; margin-bottom: 1rem; text-decoration: none;"
      >
        2021-01-01
         – posted by   
        <a
          href="/blog/author/pmc"
        >
          Adoptium PMC
        </a>
         
        <img
          alt="Adoptium PMC"
          image="[object Object]"
          stlye="[object Object]"
          style="margin-left: 0.5rem; margin-bottom: 0px; min-width: 30px; border-radius: 100%;"
        />
      </div>
    </header>
    <section>
      <p>
        Test Description
         
        <a
          href="/test-post"
        >
          Read more
        </a>
      </p>
    </section>
  </article>
</div>
`;

exports[`ArticlePreview component > should render correctly no match 1`] = `
<div>
  <article
    class="pb-3"
  >
    <header>
      <h2
        class="h3"
      >
        <a
          href="/test-post"
        >
          Test Title
        </a>
      </h2>
      <div
        class="byline"
        style="display: flex; align-items: center; margin-top: 1rem; margin-bottom: 1rem; text-decoration: none;"
      >
        2021-01-01
         – posted by   
        <a
          href="/blog/author/foo"
        >
          Test Author
        </a>
         
      </div>
    </header>
    <section>
      <p>
        Test Excerpt
         
        <a
          href="/test-post"
        >
          Read more
        </a>
      </p>
    </section>
  </article>
</div>
`;
